### ISSUE TITLE
Can't change time value when control inside Bootstrap's "Modal" window (IE 10)

### ISSUE THREAD

**Issue by ya-github-new** [weareoutman/clockpicker#26]
When Clockpicker placed **inside** "Modal" control, user can't change time value. Dropdown with dial shows  correct, user can select new hour and minute, selected values are show in dropdown, But, after click on 'Done' button - nothing changes in _input text_. 
**Error occurs only in IE10.**
**UPD:** it's possible to change editor's value, when _'autoclose'_ option set to _'true'_. If 'autoclose' option not set (or set to 'false' explicit), user can't change editor's value

**Comment by CedCannes** [comment]
I have the same issue on Firefox 34.0 and Chrome 35 installed on Ubuntu 14
When Clockpicker placed inside bootstrap modal and autoclose set to true it's work.
But if autoclose is set to false, when I click on the done button the "done" function is never called as if I had clicked outside the clickpicker to close it.

here is  a fiddle to show it http://jsfiddle.net/9p61p888/

**Comment by ksidibe** [comment]
I have the same problem with Chrome 40 on Ubuntu 14.
Anybody got a fix?

**Comment by ChandanNadig** [comment]
Facing the same issue with Chrome 36+, Firefox and IE10. See this stack overflow question http://stackoverflow.com/questions/28814155/using-bootstrap-clockpicker-in-modal-dialog/28815535#28815535

**Comment by ksidibe** [comment]
@ChandanNadig, Indeed, that's what I ended up doing too (setting autoclose to true). And that works.
Thanks a lot.
I hope this issue is addressed in the next release because, in order to provide a consistant User Experience, I had to go through my app and set autoclose to true everywhere else.

**Comment by markerictamo** [comment]
Hi guys, I found a temporary solution for this issue, although it would disable the functionality to close the clockpicker on clicking on other elements outside the popover

look for this code in your plugin:

```
// Hide when clicking or tabbing on any element except the clock, input and addon
$doc.on('click.clockpicker.' + this.id + ' focusin.clockpicker.' + this.id, function(e){
     var target = $(e.target);
     if (target.closest(self.popover).length === 0 &&
         target.closest(self.addon).length === 0 &&
     target.closest(self.input).length === 0) {
             //self.hide(); - >> comment this out
     }
});
```

**Comment by dipaksawant** [comment]
Hi guys,
I'm facing same problem but with different case.
I have selecting twelvehour option set to true.
When you select time it only sets to 'pm'  like 12:00pm and not for 12:00am

I have  updated the @CedCannes fiddle here: https://jsfiddle.net/9p61p888/3/

**Comment by govindrajg** [comment]
Hi folks,
I too have the same issue with 12 hour clock. Values are not getting updated after clicking 'Done' button. If I set to autoclose:true, then it is not waiting for the user to press 'AM' or 'PM' and by default it is updating 'input' with 'PM'.

I am using Chrome version 43 in Ubuntu.

**Comment by tsujihidetaka** [comment]
I had same issue. The following works well.
https://github.com/hikirsch/clockpicker/blob/552769e7336f1e3d9c65bcd5da777cf58c2d8193/dist/bootstrap-clockpicker.js

**Comment by diracleo** [comment]
Fixed this issue and added more options with this new code: https://github.com/diracleo/bootstrap-clockpicker

**Comment by the94air** [comment]
That's a neat work. Thanks @diracleo

**Comment by amirhos712** [comment]
@diracleo Your script doesn't work with the default scripts options. the AM PM appears by default. Thanks

**Comment by CristianJojoaCabrera** [comment]
open the modal with javascript , example-> your id modal is :
<div class="modal inmodal" id="myModal" tabindex="-1" role="dialog" aria-hidden="true" data-backdrop="static" data-attentionanimation="shake" data-keyboard="false">
<div class="modal-footer">
<button type="button" class="btn btn-primary content_option_edit_exitr_event" data-dismiss="modal">Exit</button>
</div>
**delete ->data-dismiss="modal"
**add a id button or class
**javascript -> $('.content_option_edit_exitr_event').show();
this was my solution , i  hope to can  help you
